jota/src/main/java/jota/IotaAPI.java

```diff
@@ -297,83 +274,99 @@ public StoreTransactionsResponse broadcastAndStore(final String... trytes) throw
 
     /**
      * Wrapper function for findTransactions, getTrytes and transactionObjects.
-     * Returns the transactionObject of a transaction hash. The input can be a valid findTransactions input.
+     * Returns the transactionObject of a transaction hash. The input can be a list of valid addresses.
      *
-     * @param input The inputs.
+     * @param addresses The addresses.
      * @return Transactions.
      **/
-    public List<Transaction> findTransactionObjects(String[] input) {
-        FindTransactionResponse ftr = findTransactions(input, null, null, null);
+    public List<Transaction> findTransactionObjectsByAddresses(String[] addresses) throws ArgumentException {
+        List<String> addressesWithoutChecksum = new ArrayList<>();
+
+        for (String address : addresses) {
+            String addressO = Checksum.removeChecksum(address);
+            addressesWithoutChecksum.add(addressO);
+        }
+
+        FindTransactionResponse ftr = findTransactions(addressesWithoutChecksum.toArray(new String[]{}), null, null, null);
         if (ftr == null || ftr.getHashes() == null)
             return new ArrayList<>();
         // get the transaction objects of the transactions
-        return getTransactionsObjects(ftr.getHashes());
+        return findTransactionsObjectsByHashes(ftr.getHashes());
     }
 
     /**
      * Wrapper function for findTransactions, getTrytes and transactionObjects.
-     * Returns the transactionObject of a transaction hash. The input can be a valid.
-     * findTransactions input
+     * Returns the transactionObject of a transaction hash. The input can be a list of valid tags.
      *
-     * @param input The inputs.
+     * @param tags The tags.
      * @return Transactions.
      **/
-    public List<Transaction> findTransactionObjectsByBundle(String[] input) {
-        FindTransactionResponse ftr = findTransactions(null, null, null, input);
+    public List<Transaction> findTransactionObjectsByTag(String[] tags) throws ArgumentException {
```

Comment:
Why are we using string arrays here?

Reply:
Because findTransactions requires a String[].